test/parallel/test-util-callbackify.js

```diff
@@ -225,3 +225,37 @@ const values = [
     })
   );
 }
+
+{
+  // Verify that non-function inputs throw.
+  ['foo', null, undefined, false, 0, {}, Symbol(), []].forEach((value) => {
+    assert.throws(() => {
+      callbackify(value);
+    }, common.expectsError({
+      code: 'ERR_INVALID_ARG_TYPE',
+      type: TypeError,
+      message: 'The "original" argument must be of type function'
+    }));
+  });
+}
+
+{
+  async function asyncFn() {
+    return await Promise.resolve(42);
```

Comment:
Nit: the return itself is already sufficient as the return value is [implicitly wrapped in a promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function). Therefore the `await` can be removed.